nemo/collections/vlm/hf/model/hf_auto_model_for_image_text_to_text.py

```diff
@@ -0,0 +1,205 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import lightning.pytorch as pl
+import torch
+import torch.nn.functional as F
+from transformers import AutoConfig, AutoModelForImageTextToText, AutoProcessor
+
+from nemo.collections.llm import fn
+from nemo.lightning import io
+from nemo.utils import logging
+
+
+def masked_cross_entropy(logits, targets, mask=None):
+    """Cross entropy with optional mask"""
+    if mask is not None:
+        loss = F.cross_entropy(logits, targets, reduction='none')
+        return torch.mean(loss[mask == 1])
+    else:
+        return F.cross_entropy(logits, targets)
+
+
+def align_labels(logits, labels):
+    """Aligns labels to logits
+    if length of labels == logits, then will remove first item
```

Comment:
tokens 0 1 2 3
labels 0 1 2 3
logits 1 2 3 4.

in regular vlm training, we provide tokens as 0 1 2, and labels as 1 2 3, the requirements seems to be a bit different here.